Sources/Core/ObjectiveCIR.swift

```diff
@@ -74,6 +74,37 @@ func enumTypeName(propertyName: String, className: String) -> String {
     return "\(className)\(Languages.objectiveC.snakeCaseToCamelCase(propertyName))"
 }
 
+func enumIntegralType(_ values: EnumType) -> String {
```

Comment:
Can we add a comment describing the logic here around how we're selecting the underlying type and the motivation behind this code?